### ISSUE TITLE
Lambda APM Truncating Result

### ISSUE THREAD

**Issue by bradbenvenuti** [elastic/apm-agent-nodejs#929]
**Describe the bug**
Using the APM agent with Lambda, we are seeing our response body get truncated.

**To Reproduce**
1. Use this config
`const config = {
    serviceName: 'xxx',
    secretToken: 'xxx',
    serverUrl: 'xxx'
}`
2. Then call `apm.start(config);`
3. Then do 
```
async function handler(event) {
    const data = [
        {
            postId: 1,
            id: 1,
            name: 'id labore ex et quam laborum',
            email: 'Eliseo@gardner.biz',
            body: 'laudantium enim quasi est quidem magnam voluptate ipsam eos\ntempora quo necessitatibus\ndolor quam autem quasi\nreiciendis et nam sapiente accusantium'
        },
        {
            postId: 1,
            id: 2,
            name: 'quo vero reiciendis velit similique earum',
            email: 'Jayne_Kuhic@sydney.com',
            body: 'est natus enim nihil est dolore omnis voluptatem numquam\net omnis occaecati quod ullam at\nvoluptatem error expedita pariatur\nnihil sint nostrum voluptatem reiciendis et'
        },
        {
            postId: 1,
            id: 3,
            name: 'odio adipisci rerum aut animi',
            email: 'Nikita@garfield.biz',
            body:
                'quia molestiae reprehenderit quasi aspernatur\naut expedita occaecati aliquam eveniet laudantium\nomnis quibusdam delectus saepe quia accusamus maiores nam est\ncum et ducimus et vero voluptates excepturi deleniti ratione'
        },
        {
            postId: 1,
            id: 3,
            name: 'odio adipisci rerum aut animi',
            email: 'Nikita@garfield.biz',
            body:
                'quia molestiae reprehenderit quasi aspernatur\naut expedita occaecati aliquam eveniet laudantium\nomnis quibusdam delectus saepe quia accusamus maiores nam est\ncum et ducimus et vero voluptates excepturi deleniti ratione'
        }
    ];
    const response = {
        statusCode: 200,
        body: JSON.stringify(data),
        headers: {
            'Content-Type': 'application/json'
        }
    };

    return response;
}

module.exports.handler = apm.lambda(handler);
```

```$ sls invoke local -f home
Error: Content-Type of response is application/json but body is not json
    at AwsInvokeLocal.handleResult (/Users/bradbenvenuti/.nvm/versions/node/v8.10.0/lib/node_modules/serverless/lib/plugins/aws/invokeLocal/index.js:355:19)
    at AwsInvokeLocal.callback (/Users/bradbenvenuti/.nvm/versions/node/v8.10.0/lib/node_modules/serverless/lib/plugins/aws/invokeLocal/index.js:370:26)
    at <anonymous>
    at process._tickDomainCallback (internal/process/next_tick.js:228:7)`

**Expected behavior**
Should return something like: `
{
    "statusCode": 200,
    "body": [
        {
            "postId": 1,
            "id": 1,
            "name": "id labore ex et quam laborum",
            "email": "Eliseo@gardner.biz",
            "body": "laudantium enim quasi est quidem magnam voluptate ipsam eos\ntempora quo necessitatibus\ndolor quam autem quasi\nreiciendis et nam sapiente accusantium"
        },
        {
            "postId": 1,
            "id": 2,
            "name": "quo vero reiciendis velit similique earum",
            "email": "Jayne_Kuhic@sydney.com",
            "body": "est natus enim nihil est dolore omnis voluptatem numquam\net omnis occaecati quod ullam at\nvoluptatem error expedita pariatur\nnihil sint nostrum voluptatem reiciendis et"
        },
        {
            "postId": 1,
            "id": 3,
            "name": "odio adipisci rerum aut animi",
            "email": "Nikita@garfield.biz",
            "body": "quia molestiae reprehenderit quasi aspernatur\naut expedita occaecati aliquam eveniet laudantium\nomnis quibusdam delectus saepe quia accusamus maiores nam est\ncum et ducimus et vero voluptates excepturi deleniti ratione"
        }
    ],
    "headers": {
        "Content-Type": "application/json"
    }
}

```

**Environment (please complete the following information)**
- OS: [e.g. Linux]
- Node.js version:
- APM Server version:
- Agent version:

**How are you starting the agent? (please tick one of the boxes)**
- [x] Calling `agent.start()` directly (e.g. `require('elastic-apm-node').start(...)`)
- [ ] Requiring `elastic-apm-node/start` from within the source code
- [ ] Starting node with `-r elastic-apm-node/start`

**Additional context**
If I change this line: https://github.com/elastic/apm-agent-nodejs/blob/master/lib/lambda.js#L47
To create a copy of the object like `
output: { ...result }
`
then everything works as expected. So it seems like this object is being mutated somewhere.

**Comment by watson** [comment]
Thanks for reporting. I think this is a duplicate of #909 + elastic/apm-nodejs-http-client#46 which we're currently working to fix

**Comment by watson** [comment]
Marked as duplicate - Closing for now. Feel free to leave a comment if you want it to be reopened.

**Comment by watson** [comment]
This issue was fixed in v2.7.0 which was just released

**Comment by bradbenvenuti** [comment]
Great, thank you for the update!
